### PR TITLE
fix: fixed a bug in adding string columns

### PR DESCRIPTION
**Pull request by sg-s**

### CODE REVIEW

File: docs/how-to/data-hub/create.md [hunk 1]
```diff
@@ -98,7 +98,6 @@ To create a new database column in an existing database, run:
     from deeporigin.data_hub import api
     api.add_database_column(
         database_id="existing-database-id",
-        key="unique-key",
```
Comment (sg-s) [inline docs/how-to/data-hub/create.md:101]: removed key

File: src/utils/constants.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 
 DataType = Literal[
     "integer",
-    "str",
```
Comment (sg-s) [inline src/utils/constants.py:15]: this type did not exist

File: src/data_hub/dataframe.py [hunk 3]
```diff
@@ -341,7 +340,7 @@ def _constructor(self):
 
 
 @beartype
-def _infer_column_type(column: pd.Series):
+def _infer_column_type(column: pd.Series) -> DataType:
```
Comment (sg-s) [inline src/data_hub/dataframe.py:343]: we now force return to be of this type

File: src/data_hub/dataframe.py [hunk 4]
```diff
@@ -382,7 +381,7 @@ def is_date(val):
         "integer": non_null_values.apply(lambda x: is_int(x)).all(),
         "float": non_null_values.apply(lambda x: is_float(x)).all(),
         "date": non_null_values.apply(lambda x: is_date(x)).all(),
-        "string": non_null_values.apply(lambda x: isinstance(x, str)).all(),
```
Comment (sg-s) [inline src/data_hub/dataframe.py:385]: no such type

**Review by kennyjwilli — APPROVED** [review]

**Pull request merged**